loadtest/publish.py

```diff
@@ -0,0 +1,193 @@
+""" Molotov-based test.
+
+This load test will generate some loads against the publishing and deleting
+endpoints of the profiler server.
+"""
+import json
+import os
+import base64
+from molotov import scenario, setup, global_setup, teardown, global_teardown
+from molotov import set_var, get_var
+
+_API = None
+
+# This is the server endpoint to run the load tests against.
+# Please uncomment the right line before running this file or define it with
+# the environment variable API_ENDPOINT.
+# We don't specify a default server so that the user has to consciously decide
+# the server to hit.
+
+# This is your local server.
+# _API = 'http://localhost:5252'
+
+# This is the staging server.
+# _API = 'https://dev.firefoxprofiler.nonprod.cloudops.mozgcp.net'
+
+# This is the various file sizes we'll generate in the global setup.
+_FILE_SIZES = (512, 1024, 5 * 1024, 20 * 1024)
+
+
+def setup_api_endpoint():
+    """Sets up the _API global that we use in all scenarii.
+    """
+    global _API
+
+    endpoint_from_env = os.getenv('API_ENDPOINT')
+    if endpoint_from_env is not None:
+        # The environment always wins over what's defined in the file.
+        _API = endpoint_from_env
+
+    if _API is None:
+        raise ValueError(
+                'Please define one API endpoint before running the test file, '
+                'either uncommenting the right line in the file or specifying '
+                'the env variable API_ENDPOINT.'
+                )
+    print(f"We'll run tests against the endpoint {_API}.")
+
+
+@global_setup()
+def test_starts(args):
+    """ This functions is called before anything starts.
+
+    Notice that it's not a coroutine.
+
+    Here we configure various things:
+    * we configure the API endpoint we'll use in the tests.
+    * we generate the various files to be sent in the tests.
+    """
+    setup_api_endpoint()
+    files = {x: os.urandom(x * 1024) for x in _FILE_SIZES}
+    set_var("files", files)
+
+
+@setup()
+async def worker_starts(worker_id, args):
+    """ This function is called once per worker.
+
+    If it returns a mapping, it will be used with all requests.
+
+    You can add things like Authorization headers for instance,
+    by setting a "headers" key.
+    """
+    headers = dict(Accept='application/vnd.firefox-profiler+json;version=1')
+    return dict(headers=headers)
+
+
+@teardown()
+def worker_ends(worker_id):
+    """ This functions is called when the worker is done.
+
+    Notice that it's not a coroutine.
+    """
+    pass
+
+
+@global_teardown()
+def test_ends():
+    """ This functions is called when everything is done.
+
+    Notice that it's not a coroutine.
+    """
+    pass
```

Comment:
I wondered if I should keep them, because we don't use them. For now I kept them more as documentation of what we can do, in case we need it later.